homeassistant/components/cups/sensor.py

```diff
@@ -120,21 +146,111 @@ def update(self):
         """Get the latest data and updates the states."""
         self.data.update()
         self._printer = self.data.printers.get(self._name)
+        self._available = self.data.available
+
+
+class IPPSensor(Entity):
+    """Implementation of the IPPSensor.
+
+    This sensor represents the status of the printer.
+    """
+
+    def __init__(self, data, name):
+        """Initialize the sensor."""
+        self.data = data
+        self._name = name
+        self._attributes = None
+        self._available = False
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._attributes['printer-make-and-model']
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return ICON
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self._attributes is not None:
+            try:
+                return next(v for k, v in PRINTER_STATES.items()
```

Comment:
This is a very weird check.

You just want to do 

```python
key = self._attributes['printer-state']
PRINTER_STATES.get(key, key)
```

Which means you try to get the value for `key`, and if not, return `key` as default value.